File: RB101_Programming_Foundations/small_problems/Easy1/Run2/sum_of_digits.rb
# def sum(number)
#   number.to_s.chars.map(&:to_i).reduce(&:+)
# end


def sum(number)
  digits = []

  while number > 0
    number, remainder = number.divmod(10)
    digits.unshift(remainder)
  end

  digits.reduce {|sum, digit| sum + digit }
end

puts sum(23) == 5
puts sum(496) == 19
puts sum(123_456_789) == 45

File: js210_fundamentals_of_JS_Programming/problem_sets/small_problems/easy6/sum_of_digits.js
function sum(number) {
  return String(number).split('').reduce((sum, digit) => sum + parseInt(digit, 10), 0);
}

console.log(sum(23));           // 5
console.log(sum(496));          // 19
console.log(sum(123456789));    // 45
